torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -52,10 +40,10 @@ py::object PyRRef::toHere() {
     if (rref_->isPyObj()) {
       // UserRRef<py::object>::toHere() calls python_rpc_handler which acquires
       // GIL.
-      return std::static_pointer_cast<UserRRef<py::object>>(rref_)->toHere();
+      return jit::toPyObject(std::static_pointer_cast<UserRRef>(rref_)->toHere());
     } else {
       IValue value =
-          std::static_pointer_cast<UserRRef<IValue>>(rref_)->toHere();
+          std::static_pointer_cast<UserRRef>(rref_)->toHere();
 
       {
```

Comment:
Do we have unit tests that cover both of these cases?

Reply:
hmm i think we have the test cases already for pyObj and IValue cases.